framework/src/plugins/base_plugin.ts

```diff
@@ -195,29 +182,41 @@ export interface PluginCodec {
 	encodeTransaction: (transaction: TransactionJSON) => string;
 }
 
+interface PluginInitContext {
+	config: Record<string, unknown>;
+	channel: BaseChannel;
+	options: {
+		readonly dataPath: string;
+		appConfig: Omit<ApplicationConfig, 'plugins'>;
+	};
+}
+
 export abstract class BasePlugin<
-	T extends PluginOptionsWithAppConfig = PluginOptionsWithAppConfig
+	T extends PluginOptionsWithApplicationConfig = PluginOptionsWithApplicationConfig
```

Comment:
Since we made plugin `config` and `appConfig` two separate things, so this generic type needed to be changed. It should just be plugin config. If we want to really save `options` at plugin instance level then we create another attribute in the base plugin for it.